lib/generators/shopify_app/shop_model/templates/shop.rb

```diff
@@ -2,6 +2,16 @@
 class Shop < ActiveRecord::Base
   include ShopifyApp::ShopSessionStorage
 
+  # Override access_scopes= to specify how access scopes are saved
+  def access_scopes=(scopes)
```

Comment:
The other default implementations of session storage methods are on `ShopSessionStorage`. Why put these here instead of there?

Reply:
This is to account for existing apps that already have access scopes stored and don't want to use the new migration we provide. For example, if they have an `AccessScopes` table, they would override these methods as follows

```ruby
class Shop < ActiveRecord::Base
  include ShopifyApp::ShopStorage

  def access_scopes=(scopes)
    access_scopes = AccessTokens.find_or_initialize_by(self.shopify_domain)
    AccessTokens.all_scopes = scopes
    access_scopes.save!
  end

  def access_scopes
   record = AccessScopes.find_by(shopify_domain: self.shopify_domain)
   record.all_scopes
  end
end
```

Reply:
Oh no, I love that we've added this to the interface.

My question is purely about code organization; why is this code here instead of in `ShopSessionStorage` like the implementations of all the other methods (e.g. `store`, `retrieve`)